pallets/services-payment/src/lib.rs

```diff
@@ -191,11 +207,29 @@ pub mod pallet {
 
             Ok(().into())
         }
+
+        /// Set the number of block production credits for this para_id without paying for them.
+        /// Can only be called by root.
+        #[pallet::call_index(3)]
+        #[pallet::weight(T::WeightInfo::set_credits())]
+        pub fn set_collator_assignment_credits(
+            origin: OriginFor<T>,
+            para_id: ParaId,
+            free_collator_assignment_credits: u32,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+
+            Self::set_free_collator_assignment_credits(&para_id, free_collator_assignment_credits);
+
+            Ok(().into())
+        }
     }
 
     impl<T: Config> Pallet<T> {
         /// Burn a credit for the given para. Deducts one credit if possible, errors otherwise.
-        pub fn burn_free_credit_for_para(para_id: &ParaId) -> DispatchResultWithPostInfo {
+        pub fn burn_block_production_free_credit_for_para(
```

Comment:
I love how on every PR this function name is getting longer and longer :rofl:

Reply:
hopefully we wont make it larger anymore!